### PR TITLE
React Capstone Project

### PR DESCRIPTION
**Pull request by mahdimahdawi**
**This PR Contains:**
- At First, selected an API to Fetch numeric data for the API and display the data to users on Home Page.
- Defined Redux Reducer to fetch data and pass it to React component for display.
-  Defined React Route for the page navigation that is Home page and Details page.
- On the Home page display the data that comes from the API.
- When the User Click on the more information button on each country the URL of the page changes by the country name.
- In the Details page component access the ULR and filter the list of the country for a specific county.
- In the detail page the user can see the whole information about the country and the COVID-19 cases in that country.
- Added the Test for each function and component in the project.
- For Giving style and making the page responsive Css used.
- The Webapp deployed via netlify: https://singular-gumdrop-8c2943.netlify.app/

### CODE REVIEW

**Comment by thecodechaser** [comment]
@mahdimahdawi Please check my messages on slack

File: src/components/pages/home.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const HomePage = () => {
+  const { all } = useSelector((state) => state.cases);
+  const [state] = useState({
+    value: '',
+    filteredCases: [],
+  });
+  return (
+    <>
+      {
+        state.filteredCases.length > 0 ? (
```
Comment (thecodechaser) [inline src/components/pages/home.js]: - [x] You did really a great job on this project but you missed a major requirement. Please add a feature on the home page to filter countries. (You should include a search feature on the page).
image

File: README.md [hunk 2]
```diff
@@ -25,38 +21,32 @@
...
-- LinkedIn: [LinkedIn](https://linkedin.com/in/linkedinhandle)
 
 ## 🤝 Contributing
 
 Contributions, issues, and feature requests are welcome!
 
-Feel free to check the [issues page](../../issues/).
+Feel free to check the [issues page](https://github.com/mahdimahdawi/react-capstone/issues).
 
 ## Show your support
```
Comment (thecodechaser) [inline README.md:53]: - [x] Please give appropriate credit to the author [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) in the README file for the design as requested in the project requirements.

image

File: src/components/pages/home.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import CaseComponent from '../caseComponent';
+
+const HomePage = () => {
+  const { all } = useSelector((state) => state.cases);
+  const [state] = useState({
+    value: '',
+    filteredCases: [],
+  });
+  return (
+    <>
```
Comment (thecodechaser) [inline src/components/pages/home.js]: - [x] Please make sure that it should be two cards in a row as in the design to follow the design guidelines.

### Your design
image

### Behance design
image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @mahdimahdawi👋, 

### You are doing a marvelous job on the React capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



## Required Changes ♻️

- [x] Please record a video presenting the features of the project you built. The length of the video should be a maximum of 5 minutes. (Please include in the video: a description of the project, and a demo of the project features, highlighting some interesting piece of code or something you built that you are very proud of.

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mahdimahdawi** [comment]
**Require Changes Implemented**

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @mahdimahdawi     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- No linter errors
- Nice UI design 




Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**